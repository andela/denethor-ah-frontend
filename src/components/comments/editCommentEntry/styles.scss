.comment-edit-actions {
  display: flex;
  justify-content: flex-end;
  .article-comment-field__button {
    margin-left: 0;
  }
}