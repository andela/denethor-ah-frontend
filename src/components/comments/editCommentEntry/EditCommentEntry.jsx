import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { toast } from 'react-toastify';
import PropTypes from 'prop-types';
import { TextAreaInput } from '../../textarea';
import Button from '../../Button';
import './styles.scss';
import { editComment } from '../../../redux/actions/comments';

export class EditCommentEntry extends Component {
  state = {
    commentBody: ''
  }

  componentDidMount() {
    const { comment: { commentBody, id } } = this.props;
    this.setState({ commentBody, id });
  }

  handleOnChangeComment = ({ target: { value: commentBody } }) => {
    this.setState({ commentBody });
  }

  handleOnClick = async () => {
    const { location: { pathname } } = this.props;
    const articleId = pathname.slice(pathname.lastIndexOf('/') + 1);
    const { commentBody, id } = this.state;
    const { editComment } = this.props;

    try {
      await editComment(articleId, id, commentBody);
      toast.success('comment updated successfully');
    } catch (error) {
      toast.error('error updating comment')
    }
  }

  render() {
    const { commentBody }  = this.state;
    const { toggleMode } = this.props;

    return (
        <div className='comment-create-field'>
          < TextAreaInput
              className='article-comment-field' 
              value={commentBody}
              placeHolder={'Enter comment'}
              // eslint-disable-next-line jsx-a11y/no-autofocus
              autoFocus={true}
              onChange={this.handleOnChangeComment}
          />
          <div className='comment-edit-actions'>
            <Button
            className='article-comment-field__button' 
            type='submit' 
            value='submit'
            onClick={this.handleOnClick}
            />
            <Button
            className='article-comment-field__button' 
            type='button' 
            value='cancel'
            onClick={() => { toggleMode('read') }}
            />
          </div>
        </div>
    );
  }
}

EditCommentEntry.propTypes = {
  comment: PropTypes.object,
  editComment: PropTypes.func,
  id: PropTypes.string,
  toggleMode: PropTypes.func,
  location: PropTypes.object
}

export default connect(undefined, { editComment })(withRouter(EditCommentEntry));
