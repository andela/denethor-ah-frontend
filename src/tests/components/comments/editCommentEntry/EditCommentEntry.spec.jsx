import React from 'react';
import { shallow } from 'enzyme';
import { EditCommentEntry } from '../../../../components/comments/editCommentEntry/EditCommentEntry';

it('Should render static component', () => {
  const comment = {
    commentBody: 'this is comment body',
    id: 'the id'
  }
  const wrapper = shallow(<EditCommentEntry comment={comment} />);
  expect(wrapper).toMatchSnapshot();
});

it('Should call handle functions onclick', () => {
  const editComment = jest.fn();
  const toggleMode = jest.fn();
  const location = { pathname: '/bal' };
  const props = {
    toggleMode, editComment, location, comment: {
      commentBody: 'this is comment body',
      id: 'the id'
    }
  };

  const wrapper = shallow(<EditCommentEntry {...props} />);

  wrapper.find('Button').at(0).simulate('click');

  expect(editComment).toHaveBeenCalled();

  wrapper.find('Button').at(1).simulate('click');
  expect(toggleMode).toHaveBeenLastCalledWith('read');
});
