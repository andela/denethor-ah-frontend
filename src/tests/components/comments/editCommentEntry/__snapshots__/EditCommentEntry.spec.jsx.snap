// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render static component 1`] = `
<div
  className="comment-create-field"
>
  <TextAreaInput
    autoFocus={true}
    className="article-comment-field"
    onChange={[Function]}
    placeHolder="Enter comment"
    placeholder="Your message"
    value="this is comment body"
  />
  <div
    className="comment-edit-actions"
  >
    <Button
      className="article-comment-field__button"
      isRequestSent={false}
      onClick={[Function]}
      type="submit"
      value="submit"
    />
    <Button
      className="article-comment-field__button"
      isRequestSent={false}
      onClick={[Function]}
      type="button"
      value="cancel"
    />
  </div>
</div>
`;
